src/core/deploy.ts

```diff
@@ -50,6 +53,23 @@ export class Deploy {
     return this._bump;
   }
 
+  async applyOverlay(api: API, overlay: string | undefined): Promise<string | undefined> {
+    if (overlay) {
```

Comment:
Yes it would better to make the `overlay` argument mandatory indeed. I guess it's because of too many moving things around during the implementation. 

Thanks for spotting that!